pkg/blockbuilder/scheduler/config.go

```diff
@@ -11,22 +11,28 @@ import (
 )
 
 type Config struct {
-	ConsumerGroup      string        `yaml:"consumer_group"`
-	SchedulingInterval time.Duration `yaml:"kafka_monitor_interval"`
-	ConsumeInterval    time.Duration `yaml:"consume_interval"`
-	StartupObserveTime time.Duration `yaml:"startup_observe_time"`
-	JobLeaseExpiry     time.Duration `yaml:"job_lease_expiry"`
+	ConsumerGroup       string        `yaml:"consumer_group"`
+	SchedulingInterval  time.Duration `yaml:"scheduling_interval"`
+	JobSize             time.Duration `yaml:"job_size"`
+	StartupObserveTime  time.Duration `yaml:"startup_observe_time"`
+	JobLeaseExpiry      time.Duration `yaml:"job_lease_expiry"`
+	LookbackOnNoCommit  time.Duration `yaml:"lookback_on_no_commit" category:"advanced"`
+	MaxScanAge          time.Duration `yaml:"max_scan_age" category:"advanced"`
+	MaxJobsPerPartition int           `yaml:"max_jobs_per_partition" category:"advanced"`
 
 	// Config parameters defined outside the block-builder-scheduler config and are injected dynamically.
 	Kafka ingest.KafkaConfig `yaml:"-"`
 }
 
 func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.StringVar(&cfg.ConsumerGroup, "block-builder-scheduler.consumer-group", "block-builder", "The Kafka consumer group used for getting/setting commmitted offsets.")
-	f.DurationVar(&cfg.SchedulingInterval, "block-builder-scheduler.scheduling-interval", 20*time.Second, "How frequently to recompute the schedule.")
-	f.DurationVar(&cfg.ConsumeInterval, "block-builder-scheduler.consume-interval", 1*time.Hour, "Interval between consumption cycles.")
-	f.DurationVar(&cfg.StartupObserveTime, "block-builder-scheduler.startup-observe-time", 25*time.Second, "How long to observe worker state before scheduling jobs.")
-	f.DurationVar(&cfg.JobLeaseExpiry, "block-builder-scheduler.job-lease-expiry", 2*time.Minute, "How long a job lease will live for before expiring.")
+	f.DurationVar(&cfg.SchedulingInterval, "block-builder-scheduler.scheduling-interval", 1*time.Minute, "How frequently to recompute the schedule.")
+	f.DurationVar(&cfg.JobSize, "block-builder-scheduler.job-size", 1*time.Hour, "How long jobs (and therefore blocks) should be.")
+	f.DurationVar(&cfg.StartupObserveTime, "block-builder-scheduler.startup-observe-time", 1*time.Minute, "How long to observe worker state before scheduling jobs.")
+	f.DurationVar(&cfg.JobLeaseExpiry, "block-builder-scheduler.job-lease-expiry", 2*time.Minute, "How long a job lease will live before expiring.")
+	f.DurationVar(&cfg.LookbackOnNoCommit, "block-builder-scheduler.lookback-on-no-commit", 6*time.Hour, "How much to look back if a commit is not found for a partition.")
```

Comment:
These are similar. But do different things. I'm open to ideas about combining them.